core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -658,19 +661,19 @@ public StartupConsistencyCheck getStartupConsistencyCheck() {
    * @throws AccessControlException if permission checking fails
    */
   public long getFileId(AlluxioURI path) throws AccessControlException {
-    long flushCounter = AsyncJournalWriter.INVALID_FLUSH_COUNTER;
+    Mutable<Long> flushCounter = new MutableObject<>(AsyncJournalWriter.INVALID_FLUSH_COUNTER);
```

Comment:
The `closable` way is interesting, but it is also tricky. Here is what the docs say:

`Note that the close methods of resources are called in the opposite order of their creation.`

That means we always have to put the counter one first, so that it is waited for after the unlocks, right?